dbms/src/Operators/AutoPassThroughHashAggHelper.cpp

```diff
@@ -0,0 +1,493 @@
+    {
+        const auto & required_column = required_header.getByPosition(col_idx);
+        if (child_header.has(required_column.name))
+        {
+            results.push_back(
+                [col_typeid = required_column.type->getTypeId(), col_name = required_column.name](const Block & block) {
+                    return genPassThroughColumnByCopy(col_typeid, col_name, block);
+                });
+            continue;
+        }
+
+        AutoPassThroughColumnGenerator generator = nullptr;
+        for (const auto & desc : aggregate_descriptions)
+        {
+            if (desc.column_name == required_column.name)
+            {
+                const String func_name = desc.function->getName();
+                if (desc.arguments.size() == 1)
+                    LOG_DEBUG(
+                        log,
+                        "setup agg func result generator. func: {}, from type: {}, to type: {}",
+                        func_name,
+                        child_header.getByPosition(desc.arguments[0]).type->getName(),
+                        desc.function->getReturnType()->getName());
+                else
+                    LOG_DEBUG(
+                        log,
+                        "setup agg func result generator. func: {}, to type: {}",
+                        func_name,
+                        desc.function->getReturnType()->getName());
+
+                ColumnPtr new_col;
+                if (func_name == "nothing")
+                {
+                    generator = [](const Block & block) {
+                        return genPassThroughColumnForNothing(block);
+                    };
+                }
+                else if (func_name == "sum")
+                {
+                    RUNTIME_CHECK(desc.arguments.size() == 1);
+                    RUNTIME_CHECK(required_column.type->getTypeId() == desc.function->getReturnType()->getTypeId());
+                    generator = chooseGeneratorForSum(
+                        desc,
+                        required_column,
+                        child_header.getByPosition(desc.arguments[0]).type,
+                        desc.function->getReturnType());
+                }
+                else if (func_name == "count")
+                {
+                    generator = chooseGeneratorForCount(desc, child_header);
+                }
+                else if (func_name == "first_row" || func_name == "min" || func_name == "max")
+                {
+                    RUNTIME_CHECK(desc.argument_names.size() == 1);
+                    generator = [col_typeid = required_column.type->getTypeId(),
+                                 col_name = desc.argument_names[0]](const Block & block) {
+                        return genPassThroughColumnByCopy(col_typeid, col_name, block);
+                    };
+                }
+                else
+                {
+                    generator = [&](const Block & block) {
```

Comment:
Do we test the performance for this path?

Reply:
Yes, it's slow. And I'd like to only disable `auto pass through` when it use `genPassThroughColumnGeneric`